### PR TITLE
[CMake] Simplify add_swift_library

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @Rostepher @gottesmm

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -1536,45 +1536,28 @@ function(add_swift_library name)
...
-        SwiftOnoneSupport)
-  endif()
-
-  # swiftSwiftOnoneSupport does not depend on itself,
-  # obviously.
-  if("${name}" STREQUAL "swiftSwiftOnoneSupport")
-    list(REMOVE_ITEM SWIFTLIB_SWIFT_MODULE_DEPENDS
-        SwiftOnoneSupport)
+     (NOT "${SWIFTLIB_SWIFT_MODULE_DEPENDS}" STREQUAL ""))
+    list(REMOVE_ITEM SWIFTLIB_SWIFT_MODULE_DEPENDS Core SiwftOnoneSupport)
```
Comment (jrose-apple) [inline cmake/modules/AddSwift.cmake]: Typo: "SiwftOnoneSupport"

Comment (compnerd) [reply cmake/modules/AddSwift.cmake]: *sigh*  Thanks.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @Rostepher
For some context around the motivation here, I'm looking to split up the `add_swift_library` into `add_swift_library` and `add_swift_target_library` because the current thing is extremely complex and nigh upon impossible to follow.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Bleh, the precondition check isnt working ... I'm not sure why; I guess we can deal with that simplification later.

File: cmake/modules/AddSwift.cmake [hunk 2]
```diff
@@ -1606,6 +1606,98 @@ function(add_swift_library name)
...
+          list(APPEND swiftlib_swift_compile_flags_all -D_LIB)
+        endif()
+      endif()
+
+
+      # Collect architecture agnostic SDK linker flags
+      set(swiftlib_link_flags_all ${SWIFTLIB_LINK_FLAGS})
+      if(${sdk} STREQUAL IOS_SIMULATOR)
+        if(${name} STREQUAL swiftMediaPlayer)
+          # message("DISABLING AUTOLINK FOR swiftMediaPlayer")
```
Comment (gottesmm) [inline cmake/modules/AddSwift.cmake]: I am not sure if you should comment this out. I think this caused problems in the past and the logging was meant to provide info if this happened.

@zisko @Rostepher is this useful for you guys? If not, we should just delete this line.

**Review by gottesmm — APPROVED** [review]
One quick thing that I want verified by @zisko and @Rostepher. Beyond that, LGTM

**Comment by gottesmm** [comment]
Actually, sorry just just read the first commit. Sorry! I need to review the rest.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
The rest of the patches look ok to me as well.

File: cmake/modules/AddSwift.cmake [hunk 3]
```diff
@@ -1606,6 +1593,98 @@ function(add_swift_library name)
...
+        elseif(SWIFTLIB_STATIC)
+          list(APPEND swiftlib_swift_compile_flags_all -D_LIB)
+        endif()
+      endif()
+
+
+      # Collect architecture agnostic SDK linker flags
+      set(swiftlib_link_flags_all ${SWIFTLIB_LINK_FLAGS})
+      if(${sdk} STREQUAL IOS_SIMULATOR)
+        if(${name} STREQUAL swiftMediaPlayer)
```
Comment (Rostepher) [inline cmake/modules/AddSwift.cmake]: Why not collapse these `if()` commands into a single one?

```CMake
if("${sdk}" STREQUAL IOS_SIMULATOR AND "${name}" STREQUAL "swiftMediaPlayer")
  list(APPEND swiftlib_link_flags_all "-Xlinker" "-ignore_auto_link")
endif()
```

Comment (compnerd) [reply cmake/modules/AddSwift.cmake]: I like it!  I just didn't think of it.  I'll do it as a follow up.

**Pull request merged**